LoopKit/InsulinKit/DoseEntry.swift

```diff
@@ -55,6 +55,19 @@ public struct DoseEntry: TimelineValue, Equatable {
     }
 }
 
+extension DoseEntry: Equatable {
```

Comment:
It looks like end date is being ignored here. This seems like a special case of "equality".  Where is it being used, and would a special method be better to determine that kind of equality?  

Code that isn't looking for exact object equality usually considers two doses the same "dose" if they share the same sync identifier. Those checks are typically done by comparing the `syncIdentifier` field.

As for `Equatable` specifically, and the `==` operator, I think most code readers would still assume that equality would mean the two structs are identical. Like in the pumpmanager code that checks if status has been updated and whether to notify clients of changes. In those cases, a change in end date would probably be important to notify about.

Reply:
This is the code that thinks 2 boluses are not the same when it should (`StatusTableViewController` lines 240-250):

```
private var bolusState: PumpManagerStatus.BolusState = .noBolus {
    didSet {
        if oldValue != bolusState {
            // Bolus starting
            if case .inProgress = bolusState {
                bolusProgressReporter = deviceManager.pumpManager?.createBolusProgressReporter(reportingOn: DispatchQueue.main)
            }
            refreshContext.update(with: .status)
            reloadData(animated: true)
        }
    }
}
```

I'll update that code compare without using `!=`

Reply:
So should this PR be closed then?